packages/yourloops/components/carbs-and-bolus/carbs-and-bolus-cell.tsx

```diff
@@ -0,0 +1,115 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { FC, useState } from 'react'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { Theme } from '@mui/material/styles'
+import { SxProps } from '@mui/system'
+import { TypeOfCell } from './carbs-and-bolus.model'
+import { LIGHT_BORDER, MANUAL_BOLUS_COLOR, RESCUE_CARBS_COLOR, useCarbsAndBolusStyles } from './carbs-and-bolus-styles'
+import { RescueCarbsTooltip } from './rescue-carbs-tooltip'
+import { ManualBolusTooltip } from './manual-bolus-tooltip'
+import { ManualBolusAveragePerRange, RescueCarbsAveragePerRange } from 'medical-domain'
+
+interface CarbsAndBolusCellProps {
+  manualBolus: ManualBolusAveragePerRange
+  rescueCarbs: RescueCarbsAveragePerRange
+  sx?: SxProps<Theme>
+  time: string
+}
+
+export const CarbsAndBolusCell: FC<CarbsAndBolusCellProps> = (props) => {
+  const { time, manualBolus, rescueCarbs, sx } = props
+  const { classes, theme } = useCarbsAndBolusStyles()
+  const { numberOfIntakes } = rescueCarbs
+  const { numberOfInjections } = manualBolus
+  const carbsCellBackgroundColor = numberOfIntakes > 0 ? RESCUE_CARBS_COLOR : theme.palette.grey[200]
+  const bolusCellBackgroundColor = numberOfInjections > 0 ? MANUAL_BOLUS_COLOR : theme.palette.grey[200]
+  const [cellOnHover, setCellOnHover] = useState<TypeOfCell>(undefined)
+
+  const openPopover = (cellOnHover: TypeOfCell): void => {
+    setCellOnHover(cellOnHover)
+  }
+
+  const closePopover = (): void => {
+    setCellOnHover(undefined)
+  }
+
+
+  return (
+    <Box
+      width="12.5%"
+      sx={{ ...sx, borderLeft: LIGHT_BORDER }}
+      data-testid="carbs-and-bolus-cell"
+    >
+      <Typography
+        variant="caption"
+        sx={{ marginLeft: '4px' }}
+      >
+        {time}
+      </Typography>
+      <Box className={classes.cellsWrapper}>
+        <Box
+          className={classes.cell}
+          sx={{ backgroundColor: carbsCellBackgroundColor }}
+          onMouseEnter={() => openPopover(TypeOfCell.RescueCarbs)}
+          onMouseLeave={closePopover}
+        >
+          <Typography
+            variant="caption"
+            sx={{ color: '#444444' }}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Aren't we supposed to use color from mui or defined global variables for colors?